### PR TITLE
feat: add paho mqtt cpp

### PR DESCRIPTION
**Pull request by gocarlos**
Specify library name and version:  **paho-mqtt-cpp/1.0.1**

depends on https://github.com/conan-io/conan-center-index/pull/914
closes https://github.com/conan-io/conan-center-index/issues/916

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by uilianries** [comment]
At first view, it looks good.

**Comment by gocarlos** [comment]
@uilianries what to do here? `Could NOT find PahoMqttC (missing: PAHO_MQTT_C_LIBRARIES)`

**Comment by uilianries** [comment]
@gocarlos https://github.com/conan-io/conan-center-index/pull/956

File: recipes/paho-mqtt-cpp/all/test_package/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,10 @@
+cmake_minimum_required(VERSION 2.8.12)
+project(test_package)
+
+set(CMAKE_VERBOSE_MAKEFILE TRUE)
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup()
+
+add_executable(${PROJECT_NAME} test_package.cpp)
+target_link_libraries(${PROJECT_NAME} ${CONAN_LIBS})
```
Comment (uilianries) [inline recipes/paho-mqtt-cpp/all/test_package/CMakeLists.txt]:
```suggestion
target_link_libraries(${PROJECT_NAME} ${CONAN_LIBS})
set_property(TARGET ${PROJECT_NAME} PROPERTY CXX_STANDARD 11)

```

Comment (uilianries) [reply recipes/paho-mqtt-cpp/all/test_package/CMakeLists.txt]: @gocarlos you need to c++11. It's working on my container.

Comment (gocarlos) [reply recipes/paho-mqtt-cpp/all/test_package/CMakeLists.txt]: is 
```python
    def configure(self):
        minimal_cpp_standard = "11"
        if self.settings.compiler.cppstd:
            tools.check_min_cppstd(self, minimal_cpp_standard)
```

not sufficient?

File: recipes/paho-mqtt-cpp/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+    def requirements(self):
+        self.requires("paho-mqtt-c/1.3.0")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name.replace("-", ".") + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def configure(self):
```
Comment (uilianries) [inline recipes/paho-mqtt-cpp/all/conanfile.py]: configure() is duplicated ... I spent 20 min to see it :man_facepalming:

Comment (SSE4) [reply recipes/paho-mqtt-cpp/all/conanfile.py]: something we can check in hooks?

Comment (uilianries) [reply recipes/paho-mqtt-cpp/all/conanfile.py]: Yes, but I can say pylint is able to warn about it, and we already have pylint hook, but it's not enabled.

**Comment by gocarlos** [comment]
> @gocarlos #956

@uilianries 
this MR is green even though #956 is still not merged?

**Comment by danimtb** [comment]
@gocarlos I think this recipe is not using the find_package names, is it?

**Comment by gocarlos** [comment]
> @gocarlos I think this recipe is not using the find_package names, is it?

https://github.com/conan-io/conan-center-index/pull/936#issuecomment-590513522

**Comment by danimtb** [comment]
Sorry, I did not read the message. I am very confused about this then... 😕

File: recipes/paho-mqtt-cpp/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    def package(self):
+        self.copy("edl-v10", src=self._source_subfolder, dst="licenses")
+        self.copy("epl-v10", src=self._source_subfolder, dst="licenses")
+        self.copy("notice.html", src=self._source_subfolder, dst="licenses")
+        cmake = self._configure_cmake()
+        cmake.install()
+        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+
+    def package_info(self):
+        self.cpp_info.libs = tools.collect_libs(self)
```
Comment (gocarlos) [inline recipes/paho-mqtt-cpp/all/conanfile.py]: probably we need to add here `PahoMqttCpp` as cmake find package name to use `find_package(PahoMqttCpp REQUIRED)`

Comment (uilianries) [reply recipes/paho-mqtt-cpp/all/conanfile.py]: Agreed: https://github.com/eclipse/paho.mqtt.cpp/blob/381e576d6c0c67c6d40906d88b2b47b709968f69/cmake/CMakeLists.txt#L1

**Pull request merged**